dataline-workers/src/main/java/io/dataline/workers/singer/SingerTap.java

```diff
@@ -115,48 +112,42 @@ public SingerTapFactory(final String imageName,
       cmd = ArrayUtils.addAll(cmd, "--state", STATE_JSON_FILENAME);
     }
 
-    try {
-      tapProcess =
-          pbf.create(
-              jobRoot,
-              imageName,
-              cmd)
-              .redirectError(jobRoot.resolve(DefaultSyncWorker.TAP_ERR_LOG).toFile())
-              .start();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+    tapProcess =
+        pbf.create(jobRoot, imageName, cmd)
+            .redirectError(jobRoot.resolve(SingerSyncWorker.TAP_ERR_LOG).toFile())
+            .start();
 
     bufferedReader = new BufferedReader(new InputStreamReader(tapProcess.getInputStream()));
+    messageIterator = streamFactory.create(new BufferedReader(new InputStreamReader(tapProcess.getInputStream()))).iterator();
+  }
+
+  public boolean hasNext() {
```

Comment:
if this thing is an iterator and is used as an iterator why aren't we taking advantage of that interface in java and at least calling it that?

Reply:
also why are you moving from stream to iterator? i thought we had agreed that either was fine. what is changing your mind?

Reply:
discussed offline: 
* will implement interator interface
* empirically it was hard to debug this as a stream as iterators are more debug friendly in java, hence the switch to iterator.